app/__init__.py

```diff
@@ -139,4 +140,21 @@ def filter(self, record):
     application.logger_name = "nowhere"
     application.logger
 
+    # Setup profiling
```

Comment:
This method is getting pretty long now, I wonder if it's worth splitting it out into sub-functions, i.e. one for profiling, one for logging etc. Might make it easier to read?